jsonnet/main.jsonnet

```
@@ -1,12 +1,23 @@
 local etcdMixin = (import 'github.com/etcd-io/etcd/contrib/mixin/mixin.libsonnet');
 local openshiftRules = (import 'custom.libsonnet');
-local a = if std.objectHasAll(etcdMixin, 'prometheusAlerts') then etcdMixin.prometheusAlerts.groups else [];
-local r = if std.objectHasAll(etcdMixin, 'prometheusRules') then etcdMixin.prometheusRules.groups else [];
-local o = if std.objectHasAll(openshiftRules, 'prometheusRules') then openshiftRules.prometheusRules.groups else [];
+
+local alertingRules = if std.objectHasAll(etcdMixin, 'prometheusAlerts') then etcdMixin.prometheusAlerts.groups else [];
+local promRules = if std.objectHasAll(etcdMixin, 'prometheusRules') then etcdMixin.prometheusRules.groups else [];
+local openshiftAlerts = if std.objectHasAll(openshiftRules, 'prometheusRules') then openshiftRules.prometheusRules.groups else [];
+
 // Exclude rules that are either OpenShift specific or do not work for OpenShift.
 // List should be ordered.
-local rules = std.map(function(group) group { rules: std.filter(function(rule) !std.setMember(rule.alert, ['etcdHighNumberOfFailedGRPCRequests', 'etcdHighNumberOfLeaderChanges', 'etcdInsufficientMembers']), super.rules) }
-                      , a + r);
+local excludeRules = std.map(function(group) group { rules: std.filter(function(rule) !std.setMember(rule.alert, ['etcdHighNumberOfFailedGRPCRequests', 'etcdHighNumberOfLeaderChanges', 'etcdInsufficientMembers']), super.rules) }, alertingRules + promRules);
+
+// modifyRules injects runbook_url to all critical alerts on all rules.
+local modifyRules = std.map(function(group) group {
```

mega nit:
```suggestion
local modifiedRules = std.map(function(group) group {
```

`modifyRules` implies action, hence function, here we have data only.